src/components/DealsCard.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState , MouseEvent} from 'react';
+import { Box, Typography, Avatar, LinearProgress, IconButton, AvatarGroup } from '@mui/material';
+import { styled } from '@mui/system';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import { useNavigate } from 'react-router-dom';
+
+
+interface User {
+  name: string;
+  photo: string;
+}
+
+interface DealsCardProps {
+  name: string;
+  country: string;
+  assignedUsers: User[];
+  probability: number;
+  stage?: string; 
+}
+
+// Styled progress bar with color change based on probability
+const StyledLinearProgress = styled(LinearProgress)<{ value: number }>(({ value, theme }) => ({
+  height: 14,
+  borderRadius: 5,
+  backgroundColor: "#CCCCCC",
+  '& .MuiLinearProgress-bar': {
+    backgroundColor: value >= 75 ? '#075F18' : value >= 60 ? '#94C31C' : value >= 40 ? '#EBDA25' : value >= 20 ? '#FF8C00' : '#CA1D1F',
+  },
+}));
+
+const DealsCard: React.FC<DealsCardProps> = ({ name, country, assignedUsers, probability, stage}) => {
+  const navigate = useNavigate();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+
+  const handleCardClick = () => {
+    if (!anchorEl) { // Only navigate if the menu is not open
+      navigate('/deals-details'); // Ensure this path matches your routing configuration
+    }
+  };
+
+
+  return (
+    <Box
+      sx={{
+        border: '2px solid #ccc',
+        borderRadius: 4,
+        p: 2,
+        width: 180, 
+        height: 250, 
+        m: 1,
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'space-between', 
+        backgroundColor: '#FEF7FF',
+        cursor: 'pointer', 
+        position: 'relative',
```

Comment:
`        '&:hover': {
        backgroundColor: '#E8E2EA',
      },`

just a small recommendation,
if you have time try this hover affect here and if you liked it keep it :D